src/main/java/org/zalando/nakadi/controller/ExceptionHandling.java

```diff
@@ -85,4 +87,18 @@ private String generateErrorTraceId() {
         }
         throw exception.getException();
     }
+
+    @ExceptionHandler(RepositoryProblemException.class)
+    public ResponseEntity<Problem> handleRepositoryProblem(final RepositoryProblemException exception,
+                                                           final NativeWebRequest request) {
+        LOG.error("Repository problem occurred", exception);
+        return Responses.create(Response.Status.SERVICE_UNAVAILABLE, exception.getMessage(), request);
+    }
+
+    @ExceptionHandler(InconsistentStateException.class)
```

Comment:
I agree that we should use more generic approach, otherwise we will have a lot of methods like that one. For example, RepositoryProblemException and InconsistentStateException are the same to methods to handle it.